souto-app/src/components/pages/manageUsersPage.js

```diff
@@ -0,0 +1,106 @@
+import axios from "axios"
+import React, { Component } from "react"
+import Menu from "../menu"
+import config from "../../config"
+import ReactLoading from "react-loading"
+import { MobileStepper, Grid } from "@material-ui/core"
+
+const styles = {
+  progressBar: {
+    flexGrow: "1",
+    backgroundColor: "transparent",
+    paddingTop: "3vh",
+    marginRight: "0",
+  },
+  names: {
+    textAlign: "center",
+    transform: "translateY(1.7vh)",
+  },
+  userProgressInfo: {
+    marginTop: "5vh",
+  },
+  percent: {
+    transform: "translateY(1.7vh)",
+  },
+  svgLoading: {
+    display: "block",
+    width: "10vw",
+    height: "10vh",
+    margin: "auto",
+  },
+  empty: {
+    textAlign: "center",
+  },
+}
+
+export default class ManageUsersPage extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      usersProgress: [],
+      display: "block",
+    }
+  }
+
+  componentDidMount = () => {
+    axios
+      .get(
+        config.server +
+          "/board/allusers/" +
+          window.localStorage.getItem("captainBoardId")
+      )
+      .then((res) => {
+        this.setState({ usersProgress: res.data, display: "none" })
+      })
+      .catch((err) => {
+        console.log(err)
+        alert("Sorry could not get the data please try again later ")
+      })
+  }
+
+  render() {
+    return (
+      <div>
+        <Menu isManager={true} title="My crew progress" />
+        <Grid style={styles.userProgressInfo} container spacing={1}>
+          <Grid style={{ display: `${this.state.display}` }} item xs={12}>
```

Comment:
Maybe lets put the loading in the middle of the screen and have mateiral backdrop?